bodhi/server/validators.py

```diff
@@ -1232,3 +1261,46 @@ def validate_request(request, **kwargs):
                         build.package.name, build.evr, target.description, other_build.evr))
                 request.errors.status = HTTPBadRequest.code
                 return
+
+
+@postschema_validator
+def validate_from_tag(request: pyramid.request.Request, **kwargs: dict):
+    """Check that an existing from_tag is valid and set `builds`.
+
+    Ensure that `from_tag` is a valid Koji tag and set `builds` to latest
+    builds from this tag if unset.
+
+    Args:
+        request: The current request.
+        kwargs: The kwargs of the related service definition. Unused.
+    """
+    koji_tag = request.validated.get('from_tag')
+
+    if koji_tag:
+        koji_client = buildsys.get_session()
+
+        if request.validated.get('builds'):
+            # Verify that the tag exists in Koji.
+            if not koji_client.getTag(koji_tag):
+                request.errors.add('body', 'from_tag', "The supplied from_tag doesn't exist.")
+                return
+            request.validated['builds_from_tag'] = False
+        else:
+            # Pulling the list of latest NVRs in a tag here is necessary for
+            # validation of ACLs pertaining the respective components.
+            try:
+                request.validated['builds'] = [
+                    b['nvr'] for b in koji_client.listTagged(koji_tag, latest=True)
+                ]
+            except koji.GenericError as e:
+                if "invalid taginfo" in str(e).lower():
+                    request.errors.add('body', 'from_tag', "The supplied from_tag doesn't exist.")
+                else:
+                    raise
```

Comment:
Should we raise a BodhiException here ?

Reply:
Sounds like a good idea.

Reply:
Probably like this because we're good Python 3 citizens :wink::

```
    raise BodhiException(...) from e
```